doc/source/tune/guides/plot_tune-usage.rst

```diff
@@ -0,0 +1,375 @@
+    tune.run(trainable, num_samples=10, resources_per_trial={"cpu": 2})
+
+    # If you have 4 CPUs on your machine, this will run 1 trial at a time.
+    tune.run(trainable, num_samples=10, resources_per_trial={"cpu": 4})
+
+    # Fractional values are also supported, (i.e., {"cpu": 0.5}).
+    tune.run(trainable, num_samples=10, resources_per_trial={"cpu": 0.5})
+
+To leverage GPUs, you must set ``gpu`` in ``resources_per_trial``. This will automatically set ``CUDA_VISIBLE_DEVICES`` for each trial.
+
+.. code-block:: python
+
+    # If you have 8 GPUs, this will run 8 trials at once.
+    tune.run(trainable, num_samples=10, resources_per_trial={"gpu": 1})
+
+    # If you have 4 CPUs on your machine and 1 GPU, this will run 1 trial at a time.
+    tune.run(trainable, num_samples=10, resources_per_trial={"cpu": 2, "gpu": 1})
+
+You can find an example of this in the `Keras MNIST example <https://github.com/ray-project/ray/blob/master/python/ray/tune/examples/tune_mnist_keras.py>`__.
+
+.. warning:: If 'gpu' is not set, ``CUDA_VISIBLE_DEVICES`` environment variable will be set as empty, disallowing GPU access.
+
+To attach to a Ray cluster, simply run ``ray.init`` before ``tune.run``:
+
+.. code-block:: python
+
+    # Connect to an existing distributed Ray cluster
+    ray.init(address=<ray_address>)
+    tune.run(trainable, num_samples=100, resources_per_trial={"cpu": 2, "gpu": 1})
+
+Search Space (Grid/Random)
+--------------------------
+
+.. warning:: If you use a Search Algorithm, you may not be able to specify lambdas or grid search with this
+    interface, as the search algorithm may require a different search space declaration.
+
+You can specify a grid search or random search via the dict passed into ``tune.run(config=)``.
+
+.. code-block:: python
+
+    parameters = {
+        "qux": tune.sample_from(lambda spec: 2 + 2),
+        "bar": tune.grid_search([True, False]),
+        "foo": tune.grid_search([1, 2, 3]),
+        "baz": "asd",  # a constant value
+    }
+
+    tune.run(trainable, config=parameters)
+
+By default, each random variable and grid search point is sampled once. To take multiple random samples, add ``num_samples: N`` to the experiment config. If `grid_search` is provided as an argument, the grid will be repeated `num_samples` of times.
+
+.. code-block:: python
+   :emphasize-lines: 13
+
+    # num_samples=10 repeats the 3x3 grid search 10 times, for a total of 90 trials
+    tune.run(
+        my_trainable,
+        name="my_trainable",
+        config={
+            "alpha": tune.uniform(100),
+            "beta": tune.sample_from(lambda spec: spec.config.alpha * np.random.normal()),
+            "nn_layers": [
+                tune.grid_search([16, 64, 256]),
+                tune.grid_search([16, 64, 256]),
+            ],
+        },
+        num_samples=10
+    )
+
+Read about this in the :ref:`Grid/Random Search API <tune-grid-random>` page.
+
+Metrics (Auto-Filled Results)
```

Comment:
This section title is a little unclear, what is someone doing with these? Why is it important to know?